File: cpp/practices/fibonacci.cpp
#include <iostream>

int fib(int n) {
  if (n == 0)
    return 0;
  else if (n == 1)
    return 1;
  else
    return fib(n-1) + fib(n-2);
}

int main() {
  int num;
  std::cout << "Fibonacci." << "\n";
  std::cout << "Number: ";
  std::cin >> num;
  std::cout << fib(num) << "\n";
}

File: java/practice/fibonacci/fibonacci.java
class fibonacci {
  static int fibonacci(int num) {
    if (num == 0) {
      return 0;
    } else if (num == 1) {
      return 1;
    }

    return fibonacci(num-1) + fibonacci(num-2);
  }


  public static void main(String[] argv) {
    System.out.println("5th Fibonacci number");
    System.out.println(fibonacci(5));
  }
}
